test/examples/ucp_client_server.c

```diff
@@ -383,12 +451,24 @@ static char* sockaddr_get_port_str(const struct sockaddr_storage *sock_addr,
 }
 
 static int client_server_communication(ucp_worker_h worker, ucp_ep_h ep,
+                                       send_recv_type_t send_recv_type,
                                        int is_server)
 {
     int ret;
 
-    /* Client-Server communication via Stream API */
-    ret = send_recv_stream(worker, ep, is_server);
+    switch (send_recv_type) {
+    case CLIENT_SERVER_SEND_RECV_STREAM:
+        /* Client-Server communication via Stream API */
+        ret = send_recv_stream(worker, ep, is_server);
```

Comment:
i would suggest to call print_result from client_server_communication() instead of adding request_finalize() function,
and call request_wait directly from send_recv_stream() and send_recv_tag()

Reply:
@yosefe this way would need to define recv_message in client_server_communication() and pass it to the functions. imho recv_message should be used only in the send-recv functions. 
you think it should be used in client_server_communication() ?

Reply:
ok, so no need to change